[2021-11-21 20:20:12,489] {taskinstance.py:1035} INFO - Dependencies all met for <TaskInstance: DWBI_HW2_dag.extract manual__2021-11-21T19:17:56.648159+00:00 [queued]>
[2021-11-21 20:20:12,497] {taskinstance.py:1035} INFO - Dependencies all met for <TaskInstance: DWBI_HW2_dag.extract manual__2021-11-21T19:17:56.648159+00:00 [queued]>
[2021-11-21 20:20:12,497] {taskinstance.py:1241} INFO - 
--------------------------------------------------------------------------------
[2021-11-21 20:20:12,497] {taskinstance.py:1242} INFO - Starting attempt 2 of 6
[2021-11-21 20:20:12,497] {taskinstance.py:1243} INFO - 
--------------------------------------------------------------------------------
[2021-11-21 20:20:12,507] {taskinstance.py:1262} INFO - Executing <Task(_PythonDecoratedOperator): extract> on 2021-11-21 19:17:56.648159+00:00
[2021-11-21 20:20:12,510] {standard_task_runner.py:52} INFO - Started process 18201 to run task
[2021-11-21 20:20:12,513] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'DWBI_HW2_dag', 'extract', 'manual__2021-11-21T19:17:56.648159+00:00', '--job-id', '25', '--raw', '--subdir', '/home/dante/DWBI/Apache/dags/DWBI_HW2_dag.py', '--cfg-path', '/tmp/tmp36g2m_vu', '--error-file', '/tmp/tmp7mi443i4']
[2021-11-21 20:20:12,514] {standard_task_runner.py:77} INFO - Job 25: Subtask extract
[2021-11-21 20:20:12,537] {logging_mixin.py:109} INFO - Running <TaskInstance: DWBI_HW2_dag.extract manual__2021-11-21T19:17:56.648159+00:00 [running]> on host dj-XPS
[2021-11-21 20:20:12,557] {taskinstance.py:1427} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=dj
AIRFLOW_CTX_DAG_ID=DWBI_HW2_dag
AIRFLOW_CTX_TASK_ID=extract
AIRFLOW_CTX_EXECUTION_DATE=2021-11-21T19:17:56.648159+00:00
AIRFLOW_CTX_DAG_RUN_ID=manual__2021-11-21T19:17:56.648159+00:00
[2021-11-21 20:20:12,558] {logging_mixin.py:109} INFO - Successfully made contact with the database.
[2021-11-21 20:20:12,560] {logging_mixin.py:109} INFO - Successfully placed data in dataframe.
[2021-11-21 20:20:12,560] {python.py:152} INFO - Done. Returned value was:        Make     Model  ManufacturingYear   Price Horsepower EmissionCategory FuelType   VIN
0   Porsche  Panamera               2021  200000        500                E     None  None
1   Porsche   Cayenne               2019  130000        350                C     None  None
2       BMW      530d               2007   40000        313                B     None  None
3   Porsche     Macan               2018  110000        310                B     None  None
4      Audi        Q7               2015   70000        240                C     None  None
5      Audi        A6               2012   50000        272                D     None  None
6      Audi        A6               2014   60000                           D     None  None
7   Porsche  Panamera               2020  170000                           B     None  None
8   Porsche     Macan               2018   90000                           C     None  None
9       BMW      530d               2015   80000                           D     None  None
10      BMW      530d               2015                313                D     None  None
11  Porsche   Cayenne               2012                330                C     None  None
[2021-11-21 20:20:12,574] {xcom.py:333} ERROR - Could not serialize the XCom value into JSON. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your airflow config.
[2021-11-21 20:20:12,574] {taskinstance.py:1703} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1332, in _run_raw_task
    self._execute_task_with_callbacks(context)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1458, in _execute_task_with_callbacks
    result = self._execute_task(context, self.task)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1517, in _execute_task
    self.xcom_push(key=XCOM_RETURN_KEY, value=result)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/utils/session.py", line 70, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2256, in xcom_push
    XCom.set(
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/utils/session.py", line 67, in wrapper
    return func(*args, **kwargs)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/models/xcom.py", line 100, in set
    value = XCom.serialize_value(value)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/models/xcom.py", line 331, in serialize_value
    return json.dumps(value).encode('UTF-8')
  File "/home/dante/anaconda3/lib/python3.8/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/home/dante/anaconda3/lib/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/home/dante/anaconda3/lib/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/home/dante/anaconda3/lib/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type DataFrame is not JSON serializable
[2021-11-21 20:20:12,580] {taskinstance.py:1270} INFO - Marking task as UP_FOR_RETRY. dag_id=DWBI_HW2_dag, task_id=extract, execution_date=20211121T191756, start_date=20211121T192012, end_date=20211121T192012
[2021-11-21 20:20:12,591] {standard_task_runner.py:88} ERROR - Failed to execute job 25 for task extract
Traceback (most recent call last):
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/task/task_runner/standard_task_runner.py", line 85, in _start_by_fork
    args.func(args, dag=self.dag)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/cli/cli_parser.py", line 48, in command
    return func(*args, **kwargs)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/utils/cli.py", line 92, in wrapper
    return f(*args, **kwargs)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 292, in task_run
    _run_task_by_selected_method(args, dag, ti)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 107, in _run_task_by_selected_method
    _run_raw_task(args, ti)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 180, in _run_raw_task
    ti._run_raw_task(
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/utils/session.py", line 70, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1332, in _run_raw_task
    self._execute_task_with_callbacks(context)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1458, in _execute_task_with_callbacks
    result = self._execute_task(context, self.task)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1517, in _execute_task
    self.xcom_push(key=XCOM_RETURN_KEY, value=result)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/utils/session.py", line 70, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2256, in xcom_push
    XCom.set(
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/utils/session.py", line 67, in wrapper
    return func(*args, **kwargs)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/models/xcom.py", line 100, in set
    value = XCom.serialize_value(value)
  File "/home/dante/DWBI/Apache/py_env/lib/python3.8/site-packages/airflow/models/xcom.py", line 331, in serialize_value
    return json.dumps(value).encode('UTF-8')
  File "/home/dante/anaconda3/lib/python3.8/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/home/dante/anaconda3/lib/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/home/dante/anaconda3/lib/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/home/dante/anaconda3/lib/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type DataFrame is not JSON serializable
[2021-11-21 20:20:12,605] {local_task_job.py:154} INFO - Task exited with return code 1
[2021-11-21 20:20:12,623] {local_task_job.py:264} INFO - 0 downstream tasks scheduled from follow-on schedule check
